File: simple.py
import sys

def FoursumSearch(input):
    hasFoursum = False
    for i in range(0, N): # i goes through {0,...,N-1}
        for j in range(i + 1, N):
            for o in range(j + 1, N):
                for k in range(o + 1, N):
                    if (vals[i] + vals[j] + vals[o] + vals[k] == 0):
                        hasFoursum = True
                        break
    print("True" if hasFoursum else "False")


N = int(sys.stdin.readline())
vals = map(int, sys.stdin.readlines())
FoursumSearch(vals)

File: Simple.java
import java.util.Scanner;
import edu.princeton.cs.algs4.*;
import java.util.Arrays;

public class Simple{
    public static void main(String[] args) {
	     Scanner S= new Scanner(System.in);
	     int N = StdIn.readInt();
	     long[] vals = new long[N];

	     for(int i= 0; i < N; i+= 1) vals[i] = StdIn.readLong();
       /*for (int i = 0; i < 10; i++)
        FoursumSearch(vals);*/
       Stopwatch stopwatch = new Stopwatch();
       StdOut.println(FoursumSearch(vals) ? "True": "False");
       StdOut.println("Elapsed time: " + (stopwatch.elapsedTime()));
    }

    public static boolean FoursumSearch(long[] input) {
      for (int i = 0; i < input.length; i++)
        for (int j = i + 1; j < input.length; j++)
          for (int o = j + 1; o < input.length; o++)
            for (int k = o + 1; k < input.length; k++)
              if (input[i] + input[j] + input[o] + input[k] == 0)
                return true;

      //If there are no foursums.
      return false;
    }
}
